app/scripts/controller/priorityManager.js

```diff
@@ -0,0 +1,170 @@
+/**
+ * @author Hongxin Zhang on 5/12/17.
+ */
+
+/**
+ * Priorities
+ *
+ * Priority system has been evolved to tier + priority
```

Comment:
The priority system is represented with an four elements array, [tier 1 score, tier 2 score, tier 3 score, tier 4 score]. The lower the tier, the more it weights in priority.